Sources/TuistCore/MetadataProviders/XCFrameworkMetadataProvider.swift

```diff
@@ -102,54 +95,58 @@ public final class XCFrameworkMetadataProvider: PrecompiledMetadataProvider, XCF
-                let relativeArchitectureBinaryPath = binaryPath.components.suffix(3).joined(separator: "/")
-                logger
-                    .warning(
-                        "\(xcframeworkPath.basename) is missing architecture \(relativeArchitectureBinaryPath) defined in the Info.plist"
-                    )
-                return false
+        for library in libraries {
+            let hasValidArchitectures = !library.architectures.filter(archs.contains).isEmpty
+            guard hasValidArchitectures, let linking = try? linking(for: library, xcframeworkPath: xcframeworkPath) else {
+                continue
             }
-            return true
-        }) else {
-            throw XCFrameworkMetadataProviderError.supportedArchitectureReferencesNotFound(xcframeworkPath)
+
+            return linking
         }
 
-        return try path(for: library, xcframeworkPath: xcframeworkPath)
+        throw XCFrameworkMetadataProviderError.supportedArchitectureReferencesNotFound(xcframeworkPath)
     }
 
-    private func path(
+    private func linking(
         for library: XCFrameworkInfoPlist.Library,
         xcframeworkPath: AbsolutePath
-    ) throws -> AbsolutePath {
-        let binaryPath: AbsolutePath
+    ) throws -> BinaryLinking {
+        let (binaryPath, linking): (AbsolutePath, BinaryLinking?)
 
         switch library.path.extension {
         case "framework":
             binaryPath = try AbsolutePath(validating: library.identifier, relativeTo: xcframeworkPath)
                 .appending(try RelativePath(validating: library.path.pathString))
                 .appending(component: library.path.basenameWithoutExt)
+            linking = try? self.linking(binaryPath: binaryPath)
         case "a":
             binaryPath = try AbsolutePath(validating: library.identifier, relativeTo: xcframeworkPath)
                 .appending(try RelativePath(validating: library.path.pathString))
+            linking = .static
+        case "dylib":
+            binaryPath = try AbsolutePath(validating: library.identifier, relativeTo: xcframeworkPath)
+                .appending(try RelativePath(validating: library.path.pathString))
+            linking = .dynamic
         default:
             throw XCFrameworkMetadataProviderError.fileTypeNotRecognised(
                 file: library.path,
                 frameworkName: xcframeworkPath.basename
             )
         }
-        return binaryPath
+
+        guard fileHandler.exists(binaryPath), let linking else {
+            // The missing slice relative to the XCFramework folder. e.g ios-x86_64-simulator/Alamofire.framework/Alamofire
+            let relativeArchitectureBinaryPath = binaryPath.components.suffix(3).joined(separator: "/")
+            logger
+                .warning(
+                    "\(xcframeworkPath.basename) is missing architecture \(relativeArchitectureBinaryPath) defined in the Info.plist"
+                )
```

Comment:
The `XCFrameworkMetadataProviderError.supportedArchitectureReferencesNotFound` is output to the user. Therefore, this will be a bit redundant. I'd double-check the `description` for that error case to ensure the message is very descriptive. What do you think?

```suggestion
```

Reply:
Yes, I deleted the warning message logging at the beginning and restored it for passing the `test_loadMetadata_frameworkMissingArchitecture` test. I see it as just a warning, so it may make sense to help the user be aware of the missing binary issue. You can double-check the behavior.